crates/sui-framework/sources/coin.move

```diff
@@ -224,6 +230,35 @@ module sui::coin {
         )
     }
 
+    /// Split coin `self` into `n` coins with equal balances. If the balance is
+    /// not evenly divisible by `n`, the remainder is left in `self`. Return
+    /// newly created coins.
+    public fun split_n_return<T>(self: &mut Coin<T>, n: u64, ctx: &mut TxContext): vector<Coin<T>> {
+        assert!(n > 0, EInvalidArg);
+        assert!(n <= balance::value(&self.balance), ENotEnough);
+        let vec = vector::empty<Coin<T>>();
+        let i = 0;
+        let split_amount = balance::value(&self.balance) / n;
+        while (i < n - 1) {
+            vector::push_back(&mut vec, take(&mut self.balance, split_amount, ctx));
+            i = i + 1;
+        };
+        vec
+    }
+
+    /// Split coin `self` into `n` coins with equal balances. If the balance is
+    /// not evenly divisible by `n`, the remainder is left in `self`.
+    public entry fun split_n<T>(self: &mut Coin<T>, n: u64, ctx: &mut TxContext) {
+        let vec: vector<Coin<T>> = split_n_return(self, n, ctx);
+        let i = 0;
+        let len = vector::length(&vec);
+        while (i < len) {
```

Comment:
I could also simplify this a bit by calling `vector::length` in the `while` loop directly, but I wasn't sure if it is at all worth avoiding repeatedly calling that function.